runtime/compiler/il/J9MethodSymbol.cpp

```diff
@@ -161,6 +161,7 @@ static TR::RecognizedMethod canSkipNullChecks[] =
    TR::java_util_ArrayList_add,
    TR::java_util_ArrayList_ensureCapacity,
    TR::java_util_ArrayList_get,
+#if JAVA_SPEC_VERSION < 17
```

Comment:
Is this allowed now? At one time it was said `compiler` doesn't take version specific compilation approach.

Reply:
It's used around in the codebase, and it worked building a JDK17 driver (excluded the methods inside the if). Is there's a different/better way to check for the JDK level?

Reply:
Do you have a similar usage within `openj9/runtime/compiler`? VM code applies such mechanism widely though.

Reply:
If there are plans to port the RI String implementation to Java11, then we need to remember to change this `#if` statement again.